src/components/Discounts/Discounts.tsx

```diff
@@ -0,0 +1,346 @@
+import React, { useState, useEffect, useMemo, useCallback } from "react";
+import { useSelector } from "react-redux";
+import { useTable, useExpanded, useSortBy } from "react-table";
+import { Button, Icon } from "design-react-kit";
+import { Link } from "react-router-dom";
+import { useHistory } from "react-router-dom";
+import { Badge } from "design-react-kit";
+import { Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
+import Api from "../../api/index";
+import { Discounts } from "../../api/generated";
+import {
+  CREATE_DISCOUNT,
+  DASHBOARD,
+  EDIT_DISCOUNT
+} from "../../navigation/routes";
+import ProfileItem from "../Profile/ProfileItem";
+import { makeProductCategoriesString } from "../../utils/strings";
+import api from "../../api/index";
+
+const Discounts = () => {
+  const history = useHistory();
+  const [discounts, setDiscounts] = useState<any>([]);
+  const agreement = useSelector((state: any) => state.agreement.value);
+  const [modal, setModal] = useState(false);
+  const [selectedDiscount, setSelectedDiscount] = useState<any>();
+
+  const toggle = () => setModal(!modal);
+
+  const getDiscounts = async (agreementId: string) => {
+    const response = await Api.Discount.getDiscounts(agreement.id);
+    return response.data.items;
```

Comment:
Can you try to avoid to handle Promises? We can use our fp-ts lib this way:
```
import { tryCatch } from "fp-ts/lib/TaskEither";
await tryCatch(() => Api.Discount.getDiscounts(agreement.id), toError).map(response => response.data.items).run();
```